scheduler/src/cook/mesos/util.clj

```diff
@@ -690,12 +692,36 @@
       (throw (ex-info "There is overlap between committed and uncommitted jobs, there is something wrong!"
                       {:count-committed (count committed-jobs)
                        :count-uncommitted (count uncommitted-jobs)})))
+    (if dry-run?
+      (log/info "clear-uncommitted-jobs would delete" (count uncommitted-before) "uncommitted jobs submitted before" submitted-before)
+      (log/info "clear-uncommitted-jobs is deleting" (count uncommitted-before) "uncommitted jobs submitted before" submitted-before))
     (when-not dry-run?
-      (doseq [batch (partition-all 10 uncommitted-before)]
+      (doseq [batch (partition-all 100 uncommitted-before)]
         @(d/transact conn (mapv #(vector :db.fn/retractEntity (:db/id %))
                                 batch))))
     uncommitted-before))
 
+(defn clear-uncommitted-jobs-on-schedule
+  "Runs the clear-uncommitted-jobs on a schedule; 12 hours after starting up and every 24 hours thereafter.
+
+   If something goes wrong and a batch of jobs is submitted to Cook, but is never committed, they will slowly build up
+   and clutter the database, causing the scheduler loop to slow down due to get-pending-jobs having to read and skip them.
+
+   The cost appears to be around one second every 50k jobs.
+
+   This is a simple loop that nukes any uncommitted jobs older than a few days. It runs every 24 hours. There will be a minor
+   performance hiccough lasting around 1 minute per 1000 jobs or so as this runs and flushes."
+  [conn mesos-leadership-atom]
+  (let [age (tc/to-date (-> -3 t/days t/from-now))
```

Comment:
Shouldn't this be a variable in the chime function?

Reply:
Yes, it should. :+1:

Reply:
Shams suggested 7 days. I'm making that change too._